alns/accept/tests/test_record_to_record_travel.py

```diff
@@ -121,5 +100,59 @@ def test_exponential_threshold_update():
     # The relative worsening is plus one, so this should be accepted initially,
     # as the threshold is 5, resp. 0.5. In the second case, 1 > 0.5, so the
     # second should be rejected.
-    assert_(record_travel(rnd.RandomState(), Zero(), Zero(), One()))
-    assert_(not record_travel(rnd.RandomState(), Zero(), Zero(), One()))
+    assert_(record_travel(None, Zero(), None, One()))
+    assert_(not record_travel(None, Zero(), None, One()))
+
+
+@mark.parametrize(
+    "init_obj, start_gap, end_gap, n_iters, method",
+    [
+        (1, -1, 0, 1, "linear"),  # negative start
+        (1, 0, -1, 1, "linear"),  # negative end
+        (1, 0.5, 1, 1, "linear"),  # start < end
+        (1, 0, 0, 0, "linear"),  # non-positive n_iters
+        (1, 0, 0, 1, "nonlinear"),  # invalid method
+    ],
+)
+def test_autofit_raises_for_invalid_inputs(
+    init_obj: float,
+    start_gap: float,
+    end_gap: float,
+    n_iters: int,
+    method: str,
+):
+    with assert_raises(ValueError):
+        RecordToRecordTravel.autofit(
+            init_obj, start_gap, end_gap, n_iters, method
+        )
+
+
+@mark.parametrize(
+    "init_obj, start_gap, end_gap, n_iters, method, rrt_start, rrt_end, rrt_step",
```

Comment:
Question: do you have any advice on how to add the desired test results into this parametrized test? `rrt_xxx` are the desired outcomes. It feels a bit convoluted now.

Reply:
It is quite a lot of parameters, yes, but this is how I would do it as well. There are just a lot of parameters to test here.

The function does not seem too convoluted, so I think this is OK.